// Copyright 2012 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Author: mdw@google.com (Matt Welsh)
// Protobufs for client state tracking.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package net_instaweb;

// Protobuf representing a ClientState object.
// NEXT ID: 4
message ClientStateMsg {
  // The client ID.
  optional string client_id = 1;

  // Creation time in milliseconds since Jan 1, 1970 00:00:00 UTC.
  optional int64 create_time_ms = 2;

  // Recently-accessed URLs by this client.
  repeated string recent_urls = 3;
};
